bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/JsonStorage.java

```diff
@@ -261,23 +261,33 @@ private File getBackupFile(int age) {
     }
 
     /**
-     * Write out any outstanding data
+     * Write out any outstanding data.
+     * <p>
+     * This creates the backup copy at the same time as writing the database file. This avoids
+     * having to either rename the file later (which may leave a small window for there to
+     * be no file if the system crashes during the write process), or to copy the file when
+     * writing the backup copy (which would require a read and write, and is thus slower).
      */
     public void commitDatabase() {
         String s = mapper.toJson(map);
 
         synchronized (map) {
-            // Rename the file for backup
-            File rename = new File(file.getParent() + File.separator + BACKUP_EXTENSION,
-                    System.currentTimeMillis() + SEPARATOR + file.getName());
-            file.renameTo(rename);
-
+            // Write the database file
             try (FileOutputStream outputStream = new FileOutputStream(file, false);) {
                 outputStream.write(s.getBytes());
             } catch (Exception e) {
                 logger.error("Error writing JsonDB to {}. Cause {}.", file.getPath(), e.getMessage());
             }
 
+            // And also write the backup
+            File backup = new File(file.getParent() + File.separator + BACKUP_EXTENSION,
+                    System.currentTimeMillis() + SEPARATOR + file.getName());
+            try (FileOutputStream outputStream = new FileOutputStream(backup, false);) {
```

Comment:
As these lines for actually writing out the file are identical for `file` and `backup` (apart of this parameter, of course), could you please factor them out?